### PR TITLE
VIMC-2883: first pass at documentation overhaul

### PR DESCRIPTION
**Pull request by richfitz**
This PR is primarily concerned with the README as the main landing point for the package.  There will be more documentation work to get the vignettes updated too.

I've put down more than the usual number of people for review because writing is hard. Please propose alternative ways of phrasing what is here, or an entirely new way of hooking people in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,19 +8,78 @@
...
-* Archive the exported data set
+The user describes the inputs of their analysis, including:
+
+* SQL queries (if using databases)
+* Required R sources
+* External resource files (e.g., csv data files, Rmd files, templates)
+* Packages required to run the analysis
+* Dependencies on previously run analyses
+
+Along with a list of "artefacts" (file-based results) that they will produce.
```
Comment (weshinsley) [inline README.md]: Might as well keep the last one in the bullet points

File: README.md [hunk 2]
```diff
@@ -8,19 +8,78 @@
...
+
+The workflows we have developed here are oriented towards collaborative groups of researchers - other workflows are possible (indeed orderly is also designed to support a _decentralised_ workflow, though this has not been used in practice yet).
+
+In these projects we have a group of researchers who develop and test analyses locally.  These are developed on a [branch in git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) and then run on a centralised _staging environment_ (a duplicate of our production environment).  The code and outputs are reviewed with the help of [GitHub's "Pull requests"](https://help.github.com/en/articles/about-pull-requests) and then the reports are run on our production environment.
+
+Interaction with the remote environments is achieved using an [HTTP API](https://github.com/vimc/montagu-reporting-api) which orderly itself transparently uses, so that running reports remotely can be run [directly from R](https://vimc.github.io/orderly/reference/orderly_run_remote.html).  The remote systems also an interactive web interface that can be used to explore and download versions of analyses, as well as run new ones.
```
Comment (weshinsley) [inline README.md]: Edit one of the runs eg -> reports can be run remotely from within R
Check last sentence - The remote systems also *provide* an interactive web... ?

File: README.md [hunk 3]
```diff
@@ -8,19 +8,78 @@
...
 
-## Database schema
+## Internal database schema
 
 `orderly` has a database, which should be the preferred way of querying `orderly` from other programs.  The schema is programmatically described at [`inst/database/schema.yml`](inst/database/schema.yml) and automatically generated database documentation is available [here](https://vimc.github.io/orderly/schema).
```
Comment (weshinsley) [inline README.md]: Perhaps below, edit one of the `orderly`s -> The preferred way of querying `orderly` is via its own database.

File: vignettes/src/orderly.R [hunk 4]
```diff
@@ -264,8 +264,14 @@ plain_output(tree(path, "<root>"))
 
 ## First, create a directory within `src`.  The name is important and
 ## should not contain spaces (nor should it change as this will change
-## the key report id and you'll lose a chain of history).
-dir.create(file.path(path, "src", "new"))
+## the key report id and you'll lose a chain of history), then edit
+## the file `orrderly.yml` within that directory.
```
Comment (r-ash) [inline vignettes/src/orderly.R]: `orderly.yml`

File: README.md [hunk 5]
```diff
@@ -8,19 +8,78 @@
...
+
+1. creates a new empty directory
+2. copies over _only_ the declared file resources
+3. loads only the declared packages
+4. loads the declared R sources
+5. evaluates any sql queries to create R objects
+6. then runs then analysis
+7. verifies that the declared artefacts are produced
+
+It then stores metadata alongside the analysis including md5 hashes of all inputs and outputs, copies of data extracted from the database, a record of all R packages loaded at the end of the session, and (if using git) information about the git state (hash, branch and status).
```
Comment (r-ash) [inline README.md]: If this page is intended as the landing page for orderly (and you're expecting non-devs to read this) perhaps it is worth adding a link here to something describing what md5 hashes are and what they're for.

File: DESCRIPTION [hunk 6]
```diff
@@ -1,11 +1,23 @@
...
-Title: R support for montagu-reports
+Title: Lightweight Reproducible Reporting
 Version: 0.5.18
-Description: Order, create and store reports from R.
+Description: Order, create and store reports from R.  By defining a
+    lightweight interface around the inputs and outputs of an
+    analysis, a lot of the repetitive work for reproducible research
+    can be automated.  We define a simple format for organising and
+    describing work that facilitates collaborative reproducible
+    research and that acknowledges that all analyses are run multiple
```
Comment (r-ash) [inline DESCRIPTION]: I'd probably cut out the first 'that' on this line so the sentence would read

>We define a simple format for organising and describing work that facilitates collaborative reproducible research and acknowledges that all analyses are run multiple times over their lifespans

File: README.md (hunk 1 above)
Comment (richfitz) [reply README.md]: except it's not an input!

**Pull request merged**